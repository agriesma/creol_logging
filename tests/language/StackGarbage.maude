load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "ccglab0" |-> null, "r" |->
              null, "_" |-> null,
              Code: call( "ccglab0" ; "this" ; "m" ; emp ) ;
              get( "ccglab0" ; "r" ) ; free( "ccglab0" ) ; return ( emp ) >,
            < "m" : Method | Param: noVid, Att: "r" |-> null, "ccglab0" |->
              null, "_" |-> null,
              Code: (call( "ccglab0" ; "this" ; "m" ; emp ) ;
                      get( "ccglab0" ; "r" ) ; free( "ccglab0" ) ;
                      assign( "r" ; "+" ( "r" :: int(1) ) ) []
                      assign( "r" ; int(1) ))
              ; return ( "r" ) >,
      Ocnt: 0 > } .
endm

