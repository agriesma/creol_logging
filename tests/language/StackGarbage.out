==========================================
reduce in PROGRAM : state .
result State: {
< "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "m" : Method | Param: noVid, Att: "_" |-> null, "ccglab0" |-> null,
    "r" |-> null, Code: (assign("r" ; int(1))[]call("ccglab0" ; "this" ; "m" ;
    emp) ; get("ccglab0" ; "r") ; free("ccglab0") ; assign("r" ; "+"("r" ::
    int(1)))) ; return("r") >, < "run" : Method | Param: noVid, Att: "_" |->
    null, "ccglab0" |-> null, "r" |-> null, Code: call("ccglab0" ; "this" ; "m"
    ; emp) ; get("ccglab0" ; "r") ; free("ccglab0") ; return(emp) >,
  Ocnt: 0 >
}
Bye.
